portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -22,30 +22,45 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
-  private List<String> messages;
-
-  @Override
-  public void init() {
-    messages = new ArrayList<>();
-  }
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
+    Query query = new Query("Comment");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    
+    List<String> comments = new ArrayList<>();
+    for (Entity entity: results.asIterable()) {
+        String comment = (String) entity.getProperty("commentContent");
+        comments.add(comment);
+    }
+    
+    response.setContentType("application/json;");
     Gson gson = new Gson();
-    String json = gson.toJson(messages);
+    String json = gson.toJson(comments);
     response.getWriter().println(json);
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String comment = request.getParameter("comments-input");
```

Comment:
Something I didn't think to mention on your original PR but brought up on Alisha's: you should probably handle the case where `getParameter` returns `null`.

It's not possible to pass `null` with your current `<form>`, but it's always best to safe-guard against unexpected request formats, since the client can manipulate the page/request any way they wish.

Reply:
I added a conditional to check if comment is null, and if it is, I set comment equal to an empty string. Is this the proper way to handle this case or is there a better way to handle it?